File: consul-examples/src/main/js/io/vertx/examples/consul/consul_client_verticle.js
var ConsulClient = require("vertx-consul-js/consul_client");
var consulClient = ConsulClient.create(vertx);
consulClient.putValue("key11", "value11", function (putResult, putResult_err) {
  if (putResult_err == null) {
    console.log("KV pair saved");
    consulClient.getValue("key11", function (getResult, getResult_err) {
      if (getResult_err == null) {
        console.log("KV pair retrieved");
        console.log(getResult.value);
      } else {
        getResult_err.printStackTrace();
      }
    });
  } else {
    putResult_err.printStackTrace();
  }
});

File: consul-examples/src/main/ruby/io/vertx/examples/consul/consul_client_verticle.rb
require 'vertx-consul/consul_client'
consulClient = VertxConsul::ConsulClient.create($vertx)
consulClient.put_value("key11", "value11") { |putResult_err,putResult|
  if (putResult_err == nil)
    puts "KV pair saved"
    consulClient.get_value("key11") { |getResult_err,getResult|
      if (getResult_err == nil)
        puts "KV pair retrieved"
        puts getResult['value']
      else
        getResult_err.print_stack_trace()
      end
    }
  else
    putResult_err.print_stack_trace()
  end
}
